### PR TITLE
Check that :hook is compatible with other directives

### LINKED ISSUE #369: Using :hook and :if directives together leads to compile error
I discovered using :hook and :if directives together **on the same element** somewhat break things. I thought it might be my mistake but I guess it's not 😄  Looks like there is a bug. I didn't tried it on another, clean project.

Screen Shot 2021-05-04 at 18 27 36

Example code to reproduce it:
```elixir
defmodule MyView do
  use Surface.LiveView
  
  def render(assigns) do
    ~H"""
    <h1 :hook="SomeHook" :if={{ true }}>Boom</h2>
    """
  end
end
```

Here is the stacktrace
```elixir
Compiling 1 file (.ex)

== Compilation error in file lib/omnie_web/live/components/modal/modal_component.ex ==
** (FunctionClauseError) no function clause matching in Surface.Directive.Hook.process/2    
    
    The following arguments were given to Surface.Directive.Hook.process/2:
    
        # 1
        %Surface.AST.Directive{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 18, module: nil, node_alias: nil, ...>, module: Surface.Directive.Hook, name: :hook, value: %Surface.AST.AttributeExpr{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 18, module: nil, node_alias: nil, ...>, original: "Modal", value: {{:., [generated: true], [{:__aliases__, [generated: true, alias: false], [:Surface, :TypeHandler]}, :expr_to_value!]}, [generated: true], [:hook, ":hook", ["Modal"], {{:., [generated: true], [{:__aliases__, [generated: true, alias: false], [:Keyword]}, :put_new]}, [generated: true], [[], :from, {:__MODULE__, [generated: true], Surface.TypeHandler.Hook}]}, nil, "\"Modal\""]}}}
    
        # 2
        %Surface.AST.If{children: [%Surface.AST.Tag{attributes: [%Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 16, module: nil, node_alias: nil, ...>, name: :id, type: :string, type_opts: [], value: %Surface.AST.AttributeExpr{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 16, module: nil, node_alias: nil, ...>, original: "@id", value: {{:., [generated: true], [{:__aliases__, [generated: true, alias: false], [:Surface, :TypeHandler]}, :expr_to_value!]}, [generated: true], [:string, :id, [{:@, [line: 16], [{:id, [line: 16], nil}]}], [], nil, "@id"]}}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 17, module: nil, node_alias: nil, ...>, name: :class, type: :css_class, type_opts: [], value: %Surface.AST.AttributeExpr{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 17, module: nil, node_alias: nil, ...>, original: "modal-container", value: {{:., [generated: true], [{:__aliases__, [generated: true, alias: false], [:Surface, :TypeHandler]}, :expr_to_value!]}, [generated: true], [:css_class, :class, ["modal-container"], [], nil, "\"modal-container\""]}}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 19, module: nil, node_alias: nil, ...>, name: :"x-data", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "{ open: false }"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 20, module: nil, node_alias: nil, ...>, name: :"x-init", type: :string, type_opts: [], value: %Surface.AST.AttributeExpr{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 20, module: nil, node_alias: nil, ...>, original: "() => {\n    setTimeout(() => open = true, 0)\n    $watch('open', isOpen => {\n      if (!isOpen) {\n        modalHook.modalClosing({{ @leave_duration }})\n      }\n    })\n  }", value: {{:., [generated: true], [{:__aliases__, [generated: true, alias: false], [:Surface, :TypeHandler]}, :expr_to_value!]}, [generated: true], [:string, :"x-init", [{:<<>>, [line: 20], ["() => {\n    setTimeout(() => open = true, 0)\n    $watch('open', isOpen => {\n      if (!isOpen) {\n        modalHook.modalClosing(", {:"::", [line: 24], [{{:., [line: 24], [Kernel, :to_string]}, [line: 24], [{:@, [line: 24], [{:leave_duration, [line: 24], nil}]}]}, {:binary, [line: 24], Elixir}]}, ")\n      }\n    })\n  }"]}], [], nil, "() => {\n    setTimeout(() => open = true, 0)\n    $watch('open', isOpen => {\n      if (!isOpen) {\n        modalHook.modalClosing({{ @leave_duration }})\n      }\n    })\n  }"]}}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 28, module: nil, node_alias: nil, ...>, name: :"@keydown.escape.window", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "open = false"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 29, module: nil, node_alias: nil, ...>, name: :"x-show", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "open"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 30, module: nil, node_alias: nil, ...>, name: :"x-cloak", type: :string, type_opts: [], value: %Surface.AST.Literal{value: true}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 31, module: nil, node_alias: nil, ...>, name: :"aria-labelledby", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "modal-title"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 32, module: nil, node_alias: nil, ...>, name: :role, type: :string, type_opts: [], value: %Surface.AST.Literal{value: "dialog"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 33, module: nil, node_alias: nil, ...>, name: :"aria-modal", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "true"}}], children: [%Surface.AST.Literal{value: "\n  "}, %Surface.AST.Tag{attributes: [%Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 35, module: nil, node_alias: nil, ...>, name: :class, type: :css_class, type_opts: [], value: %Surface.AST.AttributeExpr{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 35, module: nil, node_alias: nil, ...>, original: "modal-inner-container", value: {{:., [generated: true], [{:__aliases__, [generated: true, alias: false], [:Surface, :TypeHandler]}, :expr_to_value!]}, [generated: true], [:css_class, :class, ["modal-inner-container"], [], nil, "\"modal-inner-container\""]}}}], children: [%Surface.AST.Literal{value: "\n    "}, %Surface.AST.Literal{value: "\n    "}, %Surface.AST.Tag{attributes: [%Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 38, module: nil, node_alias: nil, ...>, name: :"x-cloak", type: :string, type_opts: [], value: %Surface.AST.Literal{value: true}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 39, module: nil, node_alias: nil, ...>, name: :"x-show", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "open"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 40, module: nil, node_alias: nil, ...>, name: :class, type: :css_class, type_opts: [], value: %Surface.AST.AttributeExpr{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 40, module: nil, node_alias: nil, ...>, original: "modal-bg-overlay", value: {{:., [generated: true], [{:__aliases__, [generated: true, alias: false], [:Surface, :TypeHandler]}, :expr_to_value!]}, [generated: true], [:css_class, :class, ["modal-bg-overlay"], [], nil, "\"modal-bg-overlay\""]}}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 41, module: nil, node_alias: nil, ...>, name: :"x-transition:enter", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "ease-out duration-300"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 42, module: nil, node_alias: nil, ...>, name: :"x-transition:enter-start", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "opacity-0"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 43, module: nil, node_alias: nil, ...>, name: :"x-transition:enter-end", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "opacity-100"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 44, module: nil, node_alias: nil, ...>, name: :"x-transition:leave", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "ease-in duration-200"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 45, module: nil, node_alias: nil, ...>, name: :"x-transition:leave-start", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "opacity-100"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 46, module: nil, node_alias: nil, ...>, name: :"x-transition:leave-end", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "opacity-0"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 47, module: nil, node_alias: nil, ...>, name: :"aria-hidden", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "true"}}], children: [], debug: [], directives: [], element: "div", meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 37, module: nil, node_alias: nil, ...>}, %Surface.AST.Literal{value: "\n\n    "}, %Surface.AST.Literal{value: "\n    "}, %Surface.AST.Tag{attributes: [%Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 51, module: nil, node_alias: nil, ...>, name: :class, type: :css_class, type_opts: [], value: %Surface.AST.AttributeExpr{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 51, module: nil, node_alias: nil, ...>, original: "modal-alignment-trick", value: {{:., [generated: true], [{:__aliases__, [generated: true, alias: false], [:Surface, :TypeHandler]}, :expr_to_value!]}, [generated: true], [:css_class, :class, ["modal-alignment-trick"], [], nil, "\"modal-alignment-trick\""]}}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 51, module: nil, node_alias: nil, ...>, name: :"aria-hidden", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "true"}}], children: [%Surface.AST.Literal{value: "&#8203;"}], debug: [], directives: [], element: "span", meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 51, module: nil, node_alias: nil, ...>}, %Surface.AST.Literal{value: "\n\n    "}, %Surface.AST.Tag{attributes: [%Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 54, module: nil, node_alias: nil, ...>, name: :"x-cloak", type: :string, type_opts: [], value: %Surface.AST.Literal{value: true}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 55, module: nil, node_alias: nil, ...>, name: :"x-show", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "open"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 56, module: nil, node_alias: nil, ...>, name: :class, type: :css_class, type_opts: [], value: %Surface.AST.AttributeExpr{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 56, module: nil, node_alias: nil, ...>, original: "modal-content-area", value: {{:., [generated: true], [{:__aliases__, [generated: true, alias: false], [:Surface, :TypeHandler]}, :expr_to_value!]}, [generated: true], [:css_class, :class, ["modal-content-area"], [], nil, "\"modal-content-area\""]}}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 57, module: nil, node_alias: nil, ...>, name: :"x-transition:enter", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "ease-out duration-300"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 58, module: nil, node_alias: nil, ...>, name: :"x-transition:enter-start", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 59, module: nil, node_alias: nil, ...>, name: :"x-transition:enter-end", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "opacity-100 translate-y-0 sm:scale-100"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 60, module: nil, node_alias: nil, ...>, name: :"x-transition:leave", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "ease-in duration-200"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 61, module: nil, node_alias: nil, ...>, name: :"x-transition:leave-start", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "opacity-100 translate-y-0 sm:scale-100"}}, %Surface.AST.Attribute{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 62, module: nil, node_alias: nil, ...>, name: :"x-transition:leave-end", type: :string, type_opts: [], value: %Surface.AST.Literal{value: "opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"}}], children: [%Surface.AST.Literal{value: "\n      "}, %Surface.AST.Slot{default: [], directives: [], index: %Surface.AST.Literal{value: 0}, meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 64, module: nil, node_alias: nil, ...>, name: :default, props: []}, %Surface.AST.Literal{value: "\n    "}], debug: [], directives: [], element: "div", meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 53, module: nil, node_alias: nil, ...>}, %Surface.AST.Literal{value: "\n  "}], debug: [], directives: [], element: "div", meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 35, module: nil, node_alias: nil, ...>}, %Surface.AST.Literal{value: "\n"}], debug: [], directives: [%Surface.AST.Directive{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 16, module: nil, node_alias: nil, ...>, module: Surface.Directive.If, name: :if, value: %Surface.AST.AttributeExpr{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 16, module: nil, node_alias: nil, ...>, original: " false ", value: {{:., [generated: true], [{:__aliases__, [generated: true, alias: false], [:Surface, :TypeHandler]}, :expr_to_value!]}, [generated: true], [:boolean, ":if", [false], [], nil, " false "]}}}, %Surface.AST.Directive{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 18, module: nil, node_alias: nil, ...>, module: Surface.Directive.Hook, name: :hook, value: %Surface.AST.AttributeExpr{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 18, module: nil, node_alias: nil, ...>, original: "Modal", value: {{:., [generated: true], [{:__aliases__, [generated: true, alias: false], [:Surface, :TypeHandler]}, :expr_to_value!]}, [generated: true], [:hook, ":hook", ["Modal"], {{:., [generated: true], [{:__aliases__, [generated: true, alias: false], [:Keyword]}, :put_new]}, [generated: true], [[], :from, {:__MODULE__, [generated: true], Surface.TypeHandler.Hook}]}, nil, "\"Modal\""]}}}], element: "div", meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 16, module: nil, node_alias: nil, ...>}], condition: %Surface.AST.AttributeExpr{meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 16, module: nil, node_alias: nil, ...>, original: " false ", value: {{:., [generated: true], [{:__aliases__, [generated: true, alias: false], [:Surface, :TypeHandler]}, :expr_to_value!]}, [generated: true], [:boolean, ":if", [false], [], nil, " false "]}}, debug: [], meta: #Surface.AST.Meta<checks: [no_undefined_assigns: true], file: "/Users/yusufyildirim/development/omnie/lib/omnie_web/live/components/modal/modal_component.ex", line: 16, module: nil, node_alias: nil, ...>}
    
    Attempted function clauses (showing 1 out of 1):
    
        def process([22m%Surface.AST.Directive{value: %Surface.AST.AttributeExpr{value: value} = expr, meta: meta}[0m, [31m%type{attributes: attributes} = node[0m) when [31mtype === Surface.AST.Tag[0m or [31mtype === Surface.AST.VoidTag[0m
    
    (surface 0.4.0) lib/surface/directive/hook.ex:15: Surface.Directive.Hook.process/2
    (elixir 1.11.3) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
    (surface 0.4.0) lib/surface/compiler.ex:216: anonymous fn/3 in Surface.Compiler.to_ast/2
    (elixir 1.11.3) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
    (surface 0.4.0) lib/surface/compiler.ex:211: Surface.Compiler.to_ast/2
    (surface 0.4.0) lib/surface/compiler.ex:115: Surface.Compiler.compile/5
    (surface 0.4.0) expanding macro: Surface.sigil_H/2
    (omnie 0.1.0) lib/omnie_web/live/components/modal/modal_component.ex:14: OmnieWeb.Components.Modal.render/1
    (elixir 1.11.3) lib/kernel/parallel_compiler.ex:314: anonymous fn/4 in Kernel.ParallelCompiler.spawn_workers/7
```

### PR DESCRIPTION
**Pull request by Malian**
Fixes #369

### CODE REVIEW

**Review by miguel-s — APPROVED** [review]
lgtm!

Do we need tests to make sure `:hook` is also compatible with the other directived, in particular `:attrs` and `:show`?

File: test/directives_test.exs [hunk 1]
```diff
@@ -754,6 +754,32 @@ defmodule Surface.DirectivesTest do
              <div></div>
              """
     end
+
+    test "is compatible with :if directive" do
+      html =
+        render_surface do
+          ~H"""
+            <div :hook="Button" :if={{ true }}></div>
```
Comment (msaraiva) [inline test/directives_test.exs:762]:
```suggestion
          <div :hook="Button" :if={{ true }}></div>
```

File: test/directives_test.exs [hunk 2]
```diff
@@ -754,6 +754,32 @@ defmodule Surface.DirectivesTest do
...
+      assert html =~ """
+             <div phx-hook="Surface.DirectivesTest#Button"></div>
+             """
+    end
+
+    test "is compatible with :for directive" do
+      html =
+        render_surface do
+          ~H"""
+            <div :hook="Button" :for={{ _ <- [1] }}></div>
```
Comment (msaraiva) [inline test/directives_test.exs:775]:
```suggestion
          <div :hook="Button" :for={{ _ <- [1] }}></div>
```

**Comment by msaraiva** [comment]
> Do we need tests to make sure `:hook` is also compatible with the other directives, in particular `:attrs` and `:show`?

Yeah, we could have those. Just to make sure :)

**Comment by Malian** [comment]
> Do we need tests to make sure :hook is also compatible with the other directives, in particular :attrs and :show?

Done!

**Review by msaraiva — APPROVED** [review]
@Malian lovely. Thanks for addressing this! ❤️

**Pull request merged**